src/mscorlib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -154,8 +155,8 @@ internal static void Sort(T[] keys, int index, int length, Comparison<T> compare
 
         internal static int InternalBinarySearch(T[] array, int index, int length, T value, IComparer<T> comparer)
         {
-            Debug.Assert(array != null, "Check the arguments in the caller!");
-            Debug.Assert(index >= 0 && length >= 0 && (array.Length - index >= length), "Check the arguments in the caller!");
+            Contract.Requires(array != null, "Check the arguments in the caller!");
```

Comment:
Looks like incorrectly resolved conflict. Could you please undo these changes?

Reply:
Hmm ok, I erroneously submitted against a older version (Contract.Requires() was changed to Debug.Assert() throughout). Not sure what the process is; should I delete the pull request and submit another? (Fishing for advice)

Reply:
ok think I figured it out and fixed all the problems with the pull request.